src/Service/Organisation/FileCsvUploader.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Service\Organisation;
+
+use App\Entity\Organisation;
+use App\Repository\Organisation\OrganisationRepositoryInterface;
+use League\Csv\Reader;
+use League\Csv\Statement;
+use Symfony\Component\Process\Exception\LogicException;
+
+final class FileCsvUploader implements FileCsvUploaderInterface
+{
+    private $repository;
+    private $csvHeaderFormat;
+    private const ALLOWED_VALUES = [
+        0 => 'name',
+        1 => 'website',
+        2 => 'contact',
+        3 => 'comment',
+    ];
+
+    public function __construct(OrganisationRepositoryInterface $repository)
+    {
+        $this->repository = $repository;
+    }
+
+    public function read(string $pathName): Reader
+    {
+        $reader = Reader::createFromPath($pathName, 'r');
+        $records = (new Statement())->process($reader);
+        if (0 === count($records)) {
+            throw new LogicException('An empty file is not allowed');
+        }
+        if (self::ALLOWED_VALUES !== $reader->fetchOne()) {
+            throw new LogicException('Headers do not match: '.implode(', ', $reader->fetchOne()));
```

Comment:
Can we get a more precise error? (which keys are not matching, is this the number of columns or specific names, ...)